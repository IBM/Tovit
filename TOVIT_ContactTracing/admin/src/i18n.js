import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import en from './translations/en'

// the translations
// (tip move them in a JSON file and import them)
const resources = {
  en
}

i18n
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    fallbackLng: 'en',
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false // react already safes from xss
    },
    detection: {
      order: ['querystring', 'cookie', 'localStorage', 'navigator', 'htmlTag', 'path', 'subdomain'],

      // keys or params to lookup language from
      lookupQuerystring: 'lng',
      lookupCookie: 'i18next',
      lookupLocalStorage: 'i18nextLng',
      lookupFromPathIndex: 0,
      lookupFromSubdomainIndex: 0,

      // cache user language on
      caches: ['localStorage', 'cookie'],
      excludeCacheFor: ['cimode'], // languages to not persist (cookie, localStorage)

      // optional expire and domain for set cookie
      cookieMinutes: 10,
      cookieDomain: 'myDomain',

      // optional htmlTag with lang attribute, the default is:
      htmlTag: document.documentElement,

      // only detect languages that are in the whitelist
      checkWhitelist: true
    }
  })

export default i18n
