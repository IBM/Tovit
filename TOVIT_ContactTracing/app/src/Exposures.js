import React from 'react'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import GetAppIcon from '@material-ui/icons/GetApp'
import { withTheme } from '@material-ui/core/styles'
import { Translation } from 'react-i18next'
import API from './api'

const initialState = {
  exposureStatus: false,
  loaded: false,
  mode: 'default'
}

class Exposures extends React.Component {
  constructor () {
    super()
    this.state = initialState
  }

  async reset () {
    this.setState(initialState)
  }

  async downloadHistory () {
    const checkpoints = await API.exportCheckpoints()
    var dataStr = 'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(checkpoints))
    var dlAnchorElem = document.getElementById('downloadAnchorElem')
    dlAnchorElem.setAttribute('href', dataStr)
    dlAnchorElem.setAttribute('download', 'checkpoints.json')
    dlAnchorElem.click()
  }

  render () {
    const { status, statusLoaded, theme } = this.props
    const { mode } = this.state
    const statusMessageLoading = (<Translation>{t => t('statusLoadingMessage')}</Translation>)
    const statusMessageNegative = (<Translation>{t => t('statusNegativeMessage')}</Translation>)
    const statusMessagePositive = (<Translation>{t => t('statusPositiveMessage')}</Translation>)
    const riskLevelLoading = (<Translation>{t => t('statusLoadingMessage')}</Translation>)
    const riskLevelNegative = (<Translation>{t => t('standardRiskLevelMessage')}</Translation>)
    const riskLevelPositive = (<Translation>{t => t('elevatedRiskLevelMessage')}</Translation>)
    const statusMessage = statusLoaded
      ? (status
        ? statusMessagePositive
        : statusMessageNegative)
      : statusMessageLoading
    const riskLevel = statusLoaded
      ? (status
        ? riskLevelPositive
        : riskLevelNegative)
      : riskLevelLoading
    const riskLevelColor = status
      ? 'error'
      : 'primary'
    return (
      <Grid
        container
      >
        {
          ((mode === 'default') && (
            <Grid
              container
              direction='column'
              justify='center'
              alignItems='center'
            >
              <Typography style={{ marginTop: 25 }}>
                <Translation>{t => t('yourRiskLevelMessage')}</Translation>: <span style={{ color: theme.palette[riskLevelColor].main }}>{riskLevel}</span>
              </Typography>
              <Typography style={{ marginTop: 15 }}>
                {statusMessage}
              </Typography>
              <Typography style={{ marginTop: 25 }}>
                <Translation>{t => t('aboutReportMessage')}</Translation>
              </Typography>
              <Button onClick={this.downloadHistory.bind(this)} variant='contained' color='primary' aria-label='add' style={{ marginTop: 25 }}>
                <GetAppIcon />
                <Translation>{t => t('downloadHistoryButton')}</Translation>
              </Button>
              <a id='downloadAnchorElem' href='/' style={{ display: 'none' }}><Translation>{t => t('downloadHistoryButton')}</Translation></a>
            </Grid>
          ))
        }
      </Grid>
    )
  }
}

export default withTheme(Exposures)
