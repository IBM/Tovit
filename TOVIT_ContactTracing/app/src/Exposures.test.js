/* globals test, expect */
import React from 'react'
import Exposures from '../src/Exposures'
import TestRenderer from 'react-test-renderer'
import { I18nextProvider } from 'react-i18next'
import i18n from '../src/i18n'

test('Exposures component: not exposed', () => {
  const status = false
  const statusLoaded = true
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <Exposures status={status} statusLoaded={statusLoaded} />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})

test('Exposures component: exposed', () => {
  const status = true
  const statusLoaded = true
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <Exposures status={status} statusLoaded={statusLoaded} />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})
