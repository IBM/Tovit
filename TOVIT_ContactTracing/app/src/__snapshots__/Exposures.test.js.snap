// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exposures component: exposed 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      Your risk level
      : 
      <span
        style={
          Object {
            "color": "#f44336",
          }
        }
      >
        elevated
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      A possible transmission path from an infected individual to you has been discovered. You should take precautionary measures to protect yourself and others, according to the directives of the CDC  as well as local, state, and federal governments.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      If you have received a positive test, you may download your recent checkpoints below and share this file with your doctor. This will warn those who may have been exposed of their increased risk. You will remain anonymous.
    </p>
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginTop": 25,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
        Download checkpoints
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <a
      href="/"
      id="downloadAnchorElem"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Download checkpoints
    </a>
  </div>
</div>
`;

exports[`Exposures component: not exposed 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      Your risk level
      : 
      <span
        style={
          Object {
            "color": "#3f51b5",
          }
        }
      >
        standard
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      No transmission paths from infected individuals to you have been discovered at this time. However, everyone is at risk and individuals should follow the directives of the CDC as well as local, state, and federal governments.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginTop": 25,
        }
      }
    >
      If you have received a positive test, you may download your recent checkpoints below and share this file with your doctor. This will warn those who may have been exposed of their increased risk. You will remain anonymous.
    </p>
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginTop": 25,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
        Download checkpoints
      </span>
    </button>
    <a
      href="/"
      id="downloadAnchorElem"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Download checkpoints
    </a>
  </div>
</div>
`;
